### DISCUSSION TITLE
How to make CSS module applied on Server-side?

### DISCUSSION THREAD

**Discussion by ssohymind** [vercel/next.js#20846] [Help]
I'm using CSS.module for the components in my project(https://redkiwiapp.com). I'm having a problem adjusting CSS on the server-side.

If I go to Chrome developer tool -> Network -> preview, my website shows unstyled HTML.
<img width="610" alt="스크린샷 2021-01-07 오후 7 28 07" src="https://user-images.githubusercontent.com/40379222/103881964-80ff0f00-511e-11eb-95aa-1134d1f4c53f.png">
Google web cache also has unstyled HTML.

However, if I block JS, the page appears with styled HTML.. I'm not sure whether serverside CSS applied or not.

How can I adjust CSS on the server-side? I found some documents for styled-components and Material-UI(mentioning serverStyleSheet), but not for the CSS modules. **In short, Is there any additional setting I should apply for the server-side CSS?**

**Comment by Tomburgs** [comment]
What you're trying to accomplish does not make a lot of sense.

When you preview the fetched document it is not styled because the compiled css is bundled separately from the documents themselves, and the document with the necessary styles are automatically linked through `<link>` tags. This is due to multitude of reasons such as client file caching, smaller bundles, quicker document & future load times, etc.

What you want to accomplish would mean that all of the necessary styles must be included through `<style>` tags, meaning you'd have a lot of repeating content, but why? If you'd look to the left side of your screenshot, your website is styled just fine.

**Reply by ssohymind** [reply]
Thanks for the reply. 
I saw some comments that for SEO, styled HTML is important since they(for example, Google) estimate the page from the server and assume it as the page the users use. That's why I wanted to apply CSS on the server-side. I saw some websites that have exact same preview as the production version, so I'd like to know how to do it.
